### PR TITLE
Added pyright as a package.

### PR DESCRIPTION
**Pull request by thewchan**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by thewchan** [comment]
@conda-forge/staged-recipes Anything else needed for this?

**Comment by thewchan** [comment]
@conda-forge/core Sorry to ping you all here for this but not sure what else is needed here and I've pinged @conda-forge/staged-recipes a couple times already.

File: recipes/pyright/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/pyright-{{ version }}.tar.gz
+  sha256: c3c8303442ddd39003100dcc892542b1a7f38fcbb131360784a9b1ea6f8ddccf
+
+build:
+  number: 0
+  entry_points:
```
Comment (jaimergp) [inline recipes/pyright/meta.yaml:15]: `setup.py` lists [two more entry points](https://github.com/RobertCraigie/pyright-python/blob/20ba327b53f82bcab283060896fbca3d93d8a936/setup.py#L50-L53). Is that alright? Just making sure you didn't miss them on accident!

Comment (thewchan) [reply recipes/pyright/meta.yaml:15]: Not on this version; while this PR is waiting 2 new releases were made upstream; those entrypoints were added after

**Comment by jaimergp** [comment]
Thanks!

**Pull request merged**